modules/library/main/src/main/java/org/geotools/filter/function/color/HSLColor.java

```diff
@@ -0,0 +1,194 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ * 
+ *    (C) 2017, Open Source Geospatial Foundation (OSGeo)
+ *    
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.filter.function.color;
+
+import java.awt.Color;
+
+/**
+ * Support class for color functions, represents a color in HSL space (based on
+ * http://www.niwa.nu/2013/05/math-behind-colorspace-conversions-rgb-hsl/)
+ *
+ * @author Andrea Aime - GeoSolutions
+ */
+class HSLColor {
```

Comment:
Should HSLColor extend Color for consistency? Color is a normal class, with several extensions SystemColor, ColorUIResource, ...